backend/oasst_backend/prompt_repository.py

```diff
@@ -561,6 +567,14 @@ def fetch_message(self, message_id: UUID, fail_if_missing: bool = True) -> Optio
             raise OasstError("Message not found", OasstErrorCode.MESSAGE_NOT_FOUND, HTTP_404_NOT_FOUND)
         return message
 
+    def fetch_task_of_text_labels(self, message_id: UUID, fail_if_missing: bool = True) -> Optional[TextLabels]:
+        text_labels = self.db.query(TextLabels).filter(Message.id == message_id).one_or_none()
+        associated_task_ids = None
+        if text_labels is not None:
+            query = self.db.query(Task.id).join(TextLabels, Task.id == TextLabels.id).filter(Task.id == text_labels.id)
```

Comment:
These two queries could be done in one using `outerjoin()`. There can be max 1 task per text_labels entry (e.g. in this case they share the pk uuid). With outer-join the returned column value would be either None or equal to TextLabels.Id ...